### ISSUE TITLE
[BUG] Error when build with CMake a simple sample

### ISSUE THREAD

**Issue by prsolucoes** [pybind/pybind11#2676]
Hi,

I made a simple example and getting this error:
```
/usr/bin/ld: CMakeFiles/pybind11_test.dir/src/cpp/main.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
```

I see that add pthread library as CXX parameter can make it work. But i need add it manually?

I following this:
https://pybind11.readthedocs.io/en/stable/advanced/embedding.html
and
https://pybind11.readthedocs.io/en/stable/compiling.html

Link to github action run:
https://github.com/prsolucoes/pybind11-test/runs/1419948459?check_suite_focus=true

My CMakeLists.txt sample:
https://github.com/prsolucoes/pybind11-test/blob/master/CMakeLists.txt

**Comment by henryiii** [comment]
You are using threads in your cpp library:

https://github.com/prsolucoes/pybind11-test/blob/564e34fd0e3a17dfb35db1b9f7e231d261d8f32b/src/cpp/main.cpp#L4

Also, try to avoid hard-coding flags in CMake; and if you do, wrap them in nice INTERFACE targets. But you don't need to do that here, use CMake's built-in package for Threads:

```cmake
find_package(Threads)
...
target_link_libraries(${PROJECT_NAME} PRIVATE Threads::Threads)
```

Then you don't need https://github.com/prsolucoes/pybind11-test/blob/564e34fd0e3a17dfb35db1b9f7e231d261d8f32b/CMakeLists.txt#L10-L12 and it will not break if you change compilers, or try to use it from another CMakeLists.

**Comment by henryiii** [comment]
PS: before CMake 3.4, the Threads module had a bug requiring that "LANGUAGE C" also be specified even if unused. But that was fixed in 3.4, so you don't need to worry about it! :)

**Comment by prsolucoes** [comment]
Hi @henryiii,

Fixed.

Link:
https://github.com/prsolucoes/pybind11-test/blob/master/CMakeLists.txt#L32

Thanks.
